index.js

```diff
@@ -73,13 +77,20 @@ function zipBuffer (rootDir, options, callback) {
         folders[fullPath] = parentZip.folder(file);
         dive(fullPath, cb);
       } else {
-        fs.readFile(fullPath, function (err, data) {
-          if (options.each) {
-            options.each(path.join(dir, file));
-          }
-          folders[dir].file(file, data);
-          cb(err);
-        });
+        if (openFiles < maxOpenFiles)
+        {
+          openFiles++;
+          fs.readFile(fullPath, function (err, data) {
+            if (options.each) {
+              options.each(path.join(dir, file));
+            }
+            folders[dir].file(file, data);
+            openFiles--;
+            cb(err);
+          });
+        }
+        else
```

Comment:
this should have braces around to avoid future bugs (in case someone refactors it without looking too closely)